File: scan_texts.js
const fs = require('fs');
const path = require('path');

function walk(dir, fileList = []) {
  const files = fs.readdirSync(dir);
  for (const file of files) {
    const stat = fs.statSync(path.join(dir, file));
    if (stat.isDirectory()) {
      walk(path.join(dir, file), fileList);
    } else if (file.endsWith('.dart')) {
      fileList.push(path.join(dir, file));
    }
  }
  return fileList;
}

const dirs = [
  'c:\\Hansuke\\Work\\abo_glumbo_bkk\\lib',
  'c:\\Hansuke\\Work\\abo_glumbo_technician_bbk\\lib'
];

const results = [];
// Regex to match Text('something') or Text("something")
// Simplistic: Text\(\s*(['"])(.*?)\1
const regex = /Text\(\s*(['"])(.*?)\1/g;

for (const dir of dirs) {
  if (fs.existsSync(dir)) {
    const files = walk(dir);
    for (const file of files) {
      const content = fs.readFileSync(file, 'utf8');
      let match;
      while ((match = regex.exec(content)) !== null) {
        const text = match[2];
        if (!text.includes('AppLocalizations') && !text.includes('$') && text.trim().length > 0) {
           results.push(file + ': ' + text);
        }
      }
    }
  }
}

fs.writeFileSync('texts_found.txt', results.join('\n'));
console.log('Found ' + results.length + ' strings.');

File: scan_texts.py
import os
import re

def find_hardcoded_texts(directory):
    hardcoded = []
    text_pattern = re.compile(r"Text\(\s*(['\"].*?['\"])\s*[,)]")
    
    for root, dirs, files in os.walk(directory):
        for file in files:
            if file.endswith(".dart"):
                filepath = os.path.join(root, file)
                try:
                    with open(filepath, 'r', encoding='utf-8') as f:
                        content = f.read()
                        matches = text_pattern.findall(content)
                        for match in matches:
                            if not match.startswith(('\'${', '\"${')) and 'AppLocalizations' not in match:
                                hardcoded.append((filepath, match))
                except Exception as e:
                    pass
    return hardcoded

d1 = r'c:\Hansuke\Work\abo_glumbo_bkk\lib'
d2 = r'c:\Hansuke\Work\abo_glumbo_technician_bbk\lib'

with open('texts_found.txt', 'w', encoding='utf-8') as out:
    out.write('--- User App ---\n')
    for path, text in find_hardcoded_texts(d1):
        out.write(f'{path}: {text}\n')

    out.write('--- Tech App ---\n')
    for path, text in find_hardcoded_texts(d2):
        out.write(f'{path}: {text}\n')
